### ISSUE TITLE
I am trying to provide iris support for oapi-codegen,

### ISSUE THREAD

**Issue by Ns2Kracy** [kataras/iris#2138]
I am trying to provide iris support for oapi-codegen,

I think I got most of the work done but I always get `invalid memory address or nil pointer dereference` when testing.

Here is some code:

```go
func (r *RequestBuilder) GoWithHTTPHandler(t *testing.T, handler http.Handler) *CompletedRequest {
	if r.Error != nil {
		// Fail the test if we had an error
		t.Errorf("error constructing request: %s", r.Error)
		return nil
	}
	var bodyReader io.Reader
	if r.Body != nil {
		bodyReader = bytes.NewReader(r.Body)
	}

	req := httptest.NewRequest(r.Method, r.Path, bodyReader)
	for h, v := range r.Headers {
		req.Header.Add(h, v)
	}
	if host, ok := r.Headers["Host"]; ok {
		req.Host = host
	}
	for _, c := range r.Cookies {
		req.AddCookie(c)
	}

	rec := httptest.NewRecorder()

	handler.ServeHTTP(rec, req)

	return &CompletedRequest{
		Recorder: rec,
	}
}
```

```go
func doGet(t *testing.T, i *iris.Application, rawURL string) *httptest.ResponseRecorder {
	u, err := url.Parse(rawURL)
	if err != nil {
		t.Fatalf("Invalid url: %s", rawURL)
	}

	response := testutil.NewRequest().Get(u.RequestURI()).WithHost(u.Host).WithAcceptJson().GoWithIrisHandler(t, i)
	return response.Recorder
}

func TestOapiRequestValidator(t *testing.T) {
	i := iris.New()

	called := false

	// Install a request handler for /resource. We want to make sure it doesn't
	// get called.
	i.Get("/resource", func(c iris.Context) {
		called = true
	})
	// Let's send the request to the wrong server, this should fail validation
	{
		rec := doGet(t, i, "http://not.deepmap.ai/resource")
		assert.Equal(t, http.StatusBadRequest, rec.Code)
		assert.False(t, called, "Handler should not have been called")
	}
```
It will always panic on ServerHTTP.

_Originally posted by @Ns2Kracy in https://github.com/kataras/iris/discussions/2132_

**Comment by kataras** [comment]
Hello @Ns2Kracy, you have to call `i.Build()` before your tests if you don't use `i.Listen/Run`.

**Comment by Ns2Kracy** [comment]
> Hello @Ns2Kracy, you have to call `i.Build()` before your tests if you don't use `i.Listen/Run`.

Thanks, I will do my best to provide Iris support for oapi-codegen and write some examples.

**Comment by kataras** [comment]
You are welcome! Notify me here if you need any further assistance

**Comment by Ns2Kracy** [comment]
oapi-codegen has released v1.14.0 and added support for iris [v1.14.0](https://github.com/deepmap/oapi-codegen/releases/tag/v1.14.0), [Feat: Add Iris Server Support #1088](https://github.com/deepmap/oapi-codegen/pull/1088).I'll try to write some examples of oapi-codegen for iris, is there any advice.
